testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java

```diff
@@ -508,4 +508,11 @@ public void testUnionAllAboveBroadcastJoin()
     {
         assertQuery("SELECT COUNT(*) FROM region r JOIN (SELECT nationkey FROM nation UNION ALL SELECT nationkey as key FROM nation) n ON r.regionkey = n.nationkey", "VALUES 10");
     }
+
+    @Test
+    public void testDistinctAggregationWithConstantLiterals()
+    {
+        String query = "with test_distinct_agg as (select orderkey, 'asdf' constant_str from orders limit 10) select count(distinct constant_str), count(*) from test_distinct_agg";
```

Comment:
Can we simplify the query to `select count(distinct 'abc'), count(*) from orders`